### PR TITLE
implementation auth with interceptor

### PR DESCRIPTION
**Pull request by SvetlanaNagovicyna**

### CODE REVIEW

**Comment by leonxaoss** [comment]
let's convert interceptor to function.
if the user checks "remember me" we keep the session alive all the time if not - only 1 hour.

File: src/app/public/pages/login/login.component.ts [hunk 1]
```diff
@@ -51,9 +51,10 @@ export class LoginComponent implements OnInit {
 
   login(user: Omit<UserRequest, 'name'>): void {
     const rememberMe: boolean = !!this.form.value.rememberMe;
+    localStorage.setItem('rememberMe', `${rememberMe}`);
```
Comment (leonxaoss) [inline src/app/public/pages/login/login.component.ts:54]: move it to auth service

File: src/app/shared/providers/services/auth.service.ts [hunk 2]
```diff
@@ -63,8 +75,28 @@ export class AuthService {
...
   }
 
+  refreshAccessToken(): Observable<AuthResponse> {
+    const refreshToken: string | null = localStorage.getItem('refreshToken');
+
+    return this.#http.post<AuthResponse>(`${environment.fbDbUrl}/v1/token?key=${environment.apiKey}`, {refreshToken}).pipe(
+      tap((response: AuthResponse): void => {
+        localStorage.setItem('accessToken', response.idToken);
+      }),
+      catchError((error) => {
```
Comment (leonxaoss) [inline src/app/shared/providers/services/auth.service.ts:85]: you don't need to catchError here you do it in the interceptor

File: src/app/shared/providers/services/auth.service.ts [hunk 3]
```diff
@@ -63,8 +73,22 @@ export class AuthService {
...
+    }).pipe(
+      tap((response: AuthRefreshTokenResponse): void => {
+        localStorage.setItem('accessToken', response.id_token);
+      })
+    );
+  }
+
   logout(): void {
     this.setToken(null);
+    localStorage.clear();
```
Comment (leonxaoss) [inline src/app/shared/providers/services/auth.service.ts:91]: if you call setToken(null) there you call localStorage.clear(), you don't need cal it here

**Comment by SvetlanaNagovicyna** [comment]
Add to firebase rule ".write": "auth != null || newData.exists()" 
Allow entry if user is authenticated or entry is being created

**Pull request merged**